diff --git a/gcc/cp/decl.c b/gcc/cp/decl.c
--- a/gcc/cp/decl.c
+++ b/gcc/cp/decl.c
@@ -453,10 +453,10 @@
 
 /* The binding level currently in effect.  */
 
-#define current_binding_level			\
-  (cfun && cp_function_chain->bindings		\
-   ? cp_function_chain->bindings		\
-   : scope_chain->bindings)
+#define current_binding_level			\
+  (*(cfun && cp_function_chain->bindings	\
+     ? &cp_function_chain->bindings		\
+     : &scope_chain->bindings))
 
 /* The binding level of the current class, if any.  */
 
