--- a/gcc/configure
+++ b/gcc/configure
@@ -1359,7 +1359,7 @@
 if eval "test \"`echo '$''{'ac_cv_prog_cc_${ac_cc}_c_o'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
-  echo 'foo(){}' > conftest.c
+  echo 'int foo(void) { return 0; }' > conftest.c
 # Make sure it works both with $CC and with simple cc.
 # We do the test twice because some compilers refuse to overwrite an
 # existing .o file with -o, though they will create one.
